contracts/BoostedMasterChefJoe.sol

```diff
@@ -0,0 +1,474 @@
+    function init(IERC20 _dummyToken) external onlyOwner {
+        uint256 balance = _dummyToken.balanceOf(msg.sender);
+        require(balance != 0, "BoostedMasterChefJoe: Balance must exceed 0");
+        _dummyToken.safeTransferFrom(msg.sender, address(this), balance);
+        _dummyToken.approve(address(MASTER_CHEF_V2), balance);
+        MASTER_CHEF_V2.deposit(MASTER_PID, balance);
+        emit Init();
+    }
+
+    /// @notice Returns the number of BMCJ pools.
+    /// @return pools The amount of pools in this farm
+    function poolLength() external view returns (uint256 pools) {
+        pools = poolInfo.length;
+    }
+
+    /// @notice Add a new LP to the pool. Can only be called by the owner.
+    /// DO NOT add the same LP token more than once. Rewards will be messed up if you do.
+    /// @param _allocPoint AP of the new pool.
+    /// @param _lpToken Address of the LP ERC-20 token.
+    /// @param _rewarder Address of the rewarder delegate.
+    function add(
+        uint256 _allocPoint,
+        IERC20 _lpToken,
+        IRewarder _rewarder
+    ) external onlyOwner {
+        require(!lpTokens.contains(address(_lpToken)), "BoostedMasterChefJoe: LP already added");
+        require(poolInfo.length <= 50, "BoostedMasterChefJoe: Too many pools");
+        // Sanity check to ensure _lpToken is an ERC20 token
+        _lpToken.balanceOf(address(this));
+        // Sanity check if we add a rewarder
+        if (address(_rewarder) != address(0)) {
+            _rewarder.onJoeReward(address(0), 0);
+        }
+
+        uint256 lastRewardTimestamp = block.timestamp;
+        totalAllocPoint = totalAllocPoint.add(_allocPoint);
+
+        poolInfo.push(
+            PoolInfo({
+                lpToken: _lpToken,
+                allocPoint: _allocPoint,
+                lastRewardTimestamp: lastRewardTimestamp,
+                accJoePerShare: 0,
+                rewarder: _rewarder,
+                totalVeJoe: 0,
+                totalBoostedAmount: 0
+            })
+        );
+        lpTokens.add(address(_lpToken));
+        emit Add(poolInfo.length.sub(1), _allocPoint, _lpToken, _rewarder);
+    }
+
+    /// @notice Update the given pool's JOE allocation point and `IRewarder` contract. Can only be called by the owner.
+    /// @param _pid The index of the pool. See `poolInfo`
+    /// @param _allocPoint New AP of the pool
+    /// @param _rewarder Address of the rewarder delegate
+    /// @param _overwrite True if _rewarder should be `set`. Otherwise `_rewarder` is ignored
+    function set(
+        uint256 _pid,
+        uint256 _allocPoint,
+        IRewarder _rewarder,
+        bool _overwrite
+    ) external onlyOwner {
+        PoolInfo memory pool = poolInfo[_pid];
+        totalAllocPoint = totalAllocPoint.sub(poolInfo[_pid].allocPoint).add(_allocPoint);
+        pool.allocPoint = _allocPoint;
+        if (_overwrite) {
+            _rewarder.onJoeReward(address(0), 0); // sanity check
```

Comment:
Change to:
```
if (_overwrite) {
   if (address(_rewarder) != address(0)) {
      _rewarder.onJoeReward(address(0), 0);
   }
   pool.rewarder = _rewarder;
}
```

Reason is because we want to have the option to clear a rewarder.